pkg/releaseindex/release_index.go

```diff
@@ -0,0 +1,163 @@
+package releaseindex
```

Comment:
The code here is heavily inspired by https://github.com/giantswarm/e2e-harness/blob/master/pkg/framework/version_bundle.go#L66

I just tried to improve the abstraction points and not mix different words for the same thing in the API. Like authority and component are the same thing.